packages/nimble-components/src/spinner/types.ts

```diff
@@ -0,0 +1,9 @@
+export const SpinnerSize = {
+    small: 'small',
```

Comment:
I like attribute enum values giving a useful description or hint to usage to help you distinguish the cases. But I don't think this size ramp really helps with intuition on usage.

If we knew the usages and could give sizing presets for that I think it would be more valuable.
icon: 16px
toolbar: 24px;
section: 32px
full-page: 128px

Like I don't even get why 32px is the default. Is that really going to be the most used? Can we double check that is actually a useful default?

Absent of any actual meaning to the enum states, but assuming we still want them, maybe it should just be the pixel size if we have zero context for naming:

```ts
const SpinnerSize = {
    size32: undefined, // default is val undefined
    size64: '64'
}
spinner.size = SpinnerSize.size64; // keyname has `size` prefix so it can be used as an identifier
```
```html
<nimble-spinner size="64"> <!-- value is just the number as a string to use in HTML -->
```

All to say I'd strongly prefer contextual names if we can have them, but the pixel sizing instead of an arbitrary size ramp if we don't.

Reply:
We should void absolute values in the names to reduce the risk of what all will break if we change the values or overall scale of values. 

We will not be able to anticipate all potential usages of the spinner, so we should at least have some general sizes which is what the t-shirt naming provides. Information about the appropriate size for a variety of contexts can and will be provided through documentation.

Reply:
I agree that using fixed values is a bad api. I think a better api has useful context.

I disagree that under the assumption the knowledge of contexts is impossible that we should prefer t-shirt sizing under the assumption we can reliably change the sizes. We can't meaningfully change the definition of medium without knowing the context of where medium is used. We will not be able to anticipate all potential usages of medium.

Also we don't need to know all potential contexts. That is an invalid assumption. We should know the contexts that we want consistent across apps and make the enum show those so devs can use them consistently. The full page spinner size should be consistent across all apps. BUT the spinner also can be sized arbitrarily by the end application if needed.

This is the same as icons. Icons have a severity with several color states (warning, error, information, success) that should be consistent across all apps. And we make that easy for our users to make consistent by making the enum take those states. However, you can still set arbitrary colors on an icon. The states don't have to represent all possible states. Just states and contexts we expect to be consistent across apps.

Reply:
I dunno, another thought is maybe we are putting too much energy into getting the pixel perfect config. We could just rely on the percentage width and clamp the minimum. It's fundamentally just an image. Pixel perfect scaling shouldn't really be a big deal... then because we don't have good context associated presets let the apps can just configure the size. We won't offer presets until we figure out useful ones based on context and add them in the future.

Reply:
- Re-cast the size names to be more standard XS-XXL, so mapping is now  
`x-small` (16px), `small` (32px), `medium` (48px), `large` (64px), `x-large` (96px), `xx-large` (128px)
- I think 32x32 is a reasonable default as that matches our default control size (same size as default Nimble buttons/ form fields / etc). XS / 16x16 will probably also be pretty common as that size looks good alongside standard text (for example, swapping in the Nimble Spinner [here in the Test Results grid in SLE](https://dev.lifecyclesolutions.ni.com/testinsights/results) would probably be the 16x16 size, alongside the Loading text).

I somewhat agree that more descriptive size names would be good, but as Nate said docs can probably cover the gap. I'm guessing 16x16 and 32x32 will be the most common though.

Reply:
We discussed the size naming more in our team meeting today and didn't reach a consensus. The main concerns were that (1) the T-shirt size naming are new/ specific to this component / not already used elsewhere in Nimble, (2) the size-to-name mapping is somewhat arbitrary for this control only, and those size names may apply as well to other controls, and  (3) the name scheme is different than what we chose for font size tokens, which is [baseName] then Plus1/Plus2/Minus1/Minus2/etc.

For the purposes of this PR, we're
- omitting the `size` attribute
- component will scale to sizes set via CSS width/height styles (it already did this)
- default size is 16x16 (looks good alongside text, which will be a common case, it's already a use case in SLE)
- additional sizes provided by designers (32 48 64 96 128) are described on the spinner's Docs page in Storybook.